hkdf.go

```diff
@@ -109,6 +109,12 @@ func (c *hkdf1) Read(p []byte) (int, error) {
 	return n, nil
 }
 
+// maxHKDFHashLength is maximum output length of any hash used with HKDF
+const maxHKDFHashLength = 64
+
+// hkdfAllZerosSalt is a preallocated buffer of zeros used in ExtractHKDF()
+var hkdfAllZerosSalt = make([]byte, maxHKDFHashLength)
```

Comment:
It is simpler and more robust to use as array instead of an slice, like this `var hkdfAllZerosSalt [64]byte`. This way it can't be unintentionally shrank elsewhere, and we avoid defining two variables instead of one.